services/provider/server/server.go

```diff
@@ -344,6 +344,22 @@ func (s *OCSProviderServer) GetDesiredClientState(ctx context.Context, req *pb.G
 	}
 
 	klog.Infof("Found StorageConsumer for GetDesiredClientState")
+	if consumer.Status.Client != nil {
+		serverVersion, err := semver.Make(ocsVersion.Version)
+		if err != nil {
+			klog.Errorf("failed to parse server version: %v", err)
+			return nil, status.Error(codes.Internal, "malformed server version")
+		}
+		clientVersion, err := semver.Make(consumer.Status.Client.OperatorVersion)
+		if err != nil {
+			klog.Errorf("failed to parse client version: %v", err)
+			return nil, status.Error(codes.NotFound, "unable to infer client version")
+		}
+		if serverVersion.Major < clientVersion.Major || serverVersion.Minor < clientVersion.Minor {
+			klog.Errorf("client version is ahead of server version")
+			return nil, status.Error(codes.FailedPrecondition, "server can not serve the client with desired state")
+		}
+	}
```

Comment:
This code should move into a util function (of the server), and be invoked for every client related operation.
You can call it: `checkConsumerPreCondiations`